internal/InfrastructureManagerConfig.go

```diff
@@ -0,0 +1,74 @@
+package internal
+
+import (
+	"encoding/json"
+	"io"
+)
+
+type InfrastructureManagerConfig struct {
```

Comment:
When we create `config` package there will be a need to come up with some good name for this struct. According to the go convention we should avoid stuttering but it is not always possible. I think it is ok to have `config.Config`. This is short and meaningful.